gobblin-data-management/src/main/java/gobblin/data/management/copy/CopySource.java

```diff
@@ -88,57 +101,128 @@
   @Override
   public List<WorkUnit> getWorkunits(SourceState state) {
 
-    List<WorkUnit> workUnits = Lists.newArrayList();
     CopyContext copyContext = new CopyContext();
 
     try {
 
+      FileSystem sourceFs = getSourceFileSystem(state);
+      FileSystem throttledSourceFs = getOptionallyThrottledFileSystem(sourceFs, state);
```

Comment:
I think the logic for creating an (optional) thorttled fs can be in getSourceFileSystem and getTargetFileSystem